test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+﻿Describe "Tests for hashtable to PSCustomObject conversion" -Tags "CI" {
+
+    Context 'New-Object cmdlet should accept empty hashtable or $null as Property argument' {
+            
+        { $script:a = new-object psobject -property $null } | Should not throw
+        
+        It '$a should not be $null' { $script:a | Should Not Be $null }
+        It '$a type' { $script:a | should BeOfType "System.Management.automation.psobject" }
+    }
+
+    It 'New-Object cmdlet should throw terminating errors when user specifies a non-existent property or tries to assign incompatible values' {
+       
+        try
+        { 
+            $source = @"
+                        public class SampleClass6
+                        {
+                            public SampleClass6(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass6()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+            add-type -typedefinition $source
+            New-Object -TypeName SampleClass6 -Property @{aa=10;b=5}
+            Throw "Exception expected, execution should not have reached here"            
+        }
+        catch {
+            $_.Exception | Should BeOfType "InvalidOperationException"
+            $_.Exception.Message | Should Match "aa"
+            $_.FullyQualifiedErrorId | should be "InvalidOperationException,Microsoft.PowerShell.Commands.NewObjectCommand"           
+        }
+    }
+
+    Context 'Hashtable conversion to PSCustomObject succeeds (Insertion Order is not retained)' {
+        { $script:x = [pscustomobject][hashtable]@{one=1;two=2}} | Should Not  Throw
+        It '$x is not $null' { $script:x | Should Not Be $null }
+        It '$x type' { $script:x | should BeOfType "System.Management.automation.psobject" }
+   }
+       

```

Comment:
this can be just a simple it test:
```
    It 'Hashtable conversion to PSCustomObject succeeds (Insertion Order is not retained)' {
        $x = [pscustomobject][hashtable]@{one=1;two=2}
        $x | Should Not Be $null
        $x | should BeOfType "System.Management.automation.psobject"
   }
```
I'm uncomfortable with should pipelines which are not in an it block. Our guidance is also along the lines of don't test more than you need to. So the check for should not throw is superfluous if inside an it block